### PR TITLE
[h265d] VPS extraction support in DecodeHeader

### PR DESCRIPTION
**Pull request by dmitryermilov**
Signed-off-by: Dmitry Ermilov <dmitry.ermilov@intel.com>

### CODE REVIEW

File: _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp [hunk 1]
```diff
@@ -687,6 +687,23 @@ mfxStatus VideoDECODEH265::DecodeHeader(VideoCORE *core, mfxBitstream *bs, mfxVi
...
+    if (extVps)
+    {
+        RawHeader_H265 *vps = decoder.GetVPS();
+
+        if (vps->GetSize())
+        {
+            MFX_CHECK(extVps->VPSBufSize >= vps->GetSize(), MFX_ERR_NOT_ENOUGH_BUFFER);
+            extVps->VPSBufSize = (mfxU16)vps->GetSize();
+            std::copy(vps->GetPointer(), vps->GetPointer() + extVps->VPSBufSize, extVps->VPSBuffer);
+        }
```
Comment (dmitryermilov) [inline _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp:700]: @dgurulev , can you please review this part? Should we save info if vps was changed and use this info here https://github.com/Intel-Media-SDK/MediaSDK/blob/dba2964cf1953404ee5ffb17cc3fb1b079da91a5/_studio/shared/umc/codec/h265_dec/src/umc_h265_task_supplier.cpp#L2224 (as already done for SPS/PPS to close AUs)?

Comment (dgurulev) [reply _studio/mfx_lib/decode/h265/src/mfx_h265_dec_decode.cpp:700]: @dmitryermilov,

I'm not sure that VPS is permitted somewhere excepting the beginning of a bitstream, need to refer to spec.

**Comment by wangyan-intel** [comment]
What case is used this PR in so far?

**Comment by dmitryermilov** [comment]
> What case is used this PR in so far?

Sorry, I didn't get your question. Which user case? Or ?

**Comment by wangyan-intel** [comment]
> > What case is used this PR in so far?
> 
> Sorry, I didn't get your question. Which user case? Or ?

I just want to know What case this VPS extbuffer will be used in. Transcoding? Or the other use case. :)

**Comment by dmitryermilov** [comment]
> > > What case is used this PR in so far?
> > 
> > 
> > Sorry, I didn't get your question. Which user case? Or ?
> 
> I just want to know What case this VPS extbuffer will be used in. Transcoding? Or the other use case. :)

Well, it should work just from manual/API point of view. As for real usage case, some advanced apps may need to have binary forms or headers e.g. to parse and check some params of VPS/SPS/PPS. MSDK doesn't reports all of them as structures.

**Comment by wangyan-intel** [comment]
> > > > What case is used this PR in so far?
> > > 
> > > 
> > > Sorry, I didn't get your question. Which user case? Or ?
> > 
> > 
> > I just want to know What case this VPS extbuffer will be used in. Transcoding? Or the other use case. :)
> 
> Well, it should work just from manual/API point of view. As for real usage case, some advanced apps may need to have binary forms or headers e.g. to parse and check some params of VPS/SPS/PPS. MSDK doesn't reports all of them as structures.

Thanks for your clarification.

**Review by onabiull — APPROVED** [review]

**Review by dgurulev — APPROVED** [review]

**Pull request merged**